core/app/models/spree/payment_method.rb

```diff
@@ -108,9 +103,9 @@ def active?
         where(type: to_s, active: true).count > 0
       end
 
-      # @deprecated Use .with_deleted.find instead
+      # @deprecated Use .with_discarded.find instead
       def find_with_destroyed(*args)
```

Comment:
Not sure if this PR would be the best place for doing it, but I think that this deprecated method may just be removed. 
Since this PR already has many breaking changes, it may make sense to take advantage of that in order to simply remove this old code as well.